src/celeritas/track/detail/TrackInitAlgorithms.cu

```diff
@@ -46,6 +100,42 @@ size_type remove_if_alive(
 
     // New size of the vacancy vector
     return end - start;
+#else
+    // CUB functions expect a cudaStream_t pointer for the stream
+    using StreamT = CELER_DEVICE_API_SYMBOL(Stream_t);
+    StreamT stream = device().stream(stream_id).get();
+
+    DeviceVector<size_type> num_not_active{1, stream_id};
+
+    // Calling with nullptr causes the function to return the amount of working
+    // space needed instead of invoking the kernel.
+    size_t temp_storage_bytes = 0;
+    auto data = device_pointer_cast(vacancies.data());
+    // HIP defines hipCUB functions as [[nodiscard]], but we defer error checks
+    auto cub_error_code = cub::DeviceSelect::If(nullptr,
+                                                temp_storage_bytes,
+                                                data,
+                                                num_not_active.data(),
+                                                vacancies.size(),
+                                                NotNull{},
+                                                stream);
+    CELER_DISCARD(cub_error_code);
+    // Allocate temporary storage
+    DeviceVector<char> temp_storage(temp_storage_bytes, stream_id);
+    // Run selection
+    cub_error_code = cub::DeviceSelect::If(temp_storage.data(),
+                                           temp_storage_bytes,
+                                           data,
+                                           num_not_active.data(),
+                                           vacancies.size(),
+                                           NotNull{},
+                                           stream);
+    CELER_DISCARD(cub_error_code);
+    auto num = ItemCopier<size_type>{stream_id}(num_not_active.data());
```

Comment:
The copy here is asynchronous. We're not copying back to pinned memory, so cuda might execute the copy sycnhronously, but shouldn't we still have a stream synchronize here to guarantee that the host isn't accessing the memory before the device transfer is done?